From 78a8572e3ac6a6406ca50e21c3018aa103a38011 Mon Sep 17 00:00:00 2001
From: David Cohen <david.a.cohen@intel.com>
Date: Fri, 12 Jul 2013 15:35:03 -0700
Subject: [PATCH 180/429] pmic_avp: add missing intel_pmic.h header file

This missing header file is needed for pmic_avp driver. Since it's not
upstreamed at all, should not cause problems when rebasing tree.

Signed-off-by: David Cohen <david.a.cohen@intel.com>
---
 include/linux/regulator/intel_pmic.h | 54 ++++++++++++++++++++++++++++
 1 file changed, 54 insertions(+)
 create mode 100644 include/linux/regulator/intel_pmic.h

diff --git a/include/linux/regulator/intel_pmic.h b/include/linux/regulator/intel_pmic.h
new file mode 100644
index 000000000000..f972d6023542
--- /dev/null
+++ b/include/linux/regulator/intel_pmic.h
@@ -0,0 +1,54 @@
+/*
+*Support for intel pmic
+*Copyright (c) 2012, Intel Corporation.
+*This program is free software; you can redistribute it and/or modify
+*it under the terms of the GNU General Public License version 2 as
+*published by the Free Software Foundation.
+*
+*/
+
+struct regulator_init_data;
+
+enum intel_regulator_id {
+	VPROG1,
+	VPROG2,
+	VEMMC1,
+	VEMMC2,
+};
+
+/* Voltage tables for Regulators */
+static const u16 VPROG1_VSEL_table[] = {
+	1200, 1800, 2500, 2800,
+};
+
+static const u16 VPROG2_VSEL_table[] = {
+	1200, 1800, 2500, 2800,
+};
+
+static const u16 VEMMC1_VSEL_table[] = {
+	2850,
+};
+static const u16 VEMMC2_VSEL_table[] = {
+	2850,
+};
+
+static const u16 V180AON_VSEL_table[] = {
+	1800, 1817, 1836, 1854,
+};
+
+/* Slave Address for all regulators */
+#define VPROG1CNT_ADDR	0x0D6
+#define VPROG2CNT_ADDR	0x0D7
+#define VEMMC1CNT_ADDR	0x0D9
+#define VEMMC2CNT_ADDR	0x0DA
+/**
+ * intel_pmic_info - platform data for intel pmic
+ * @pmic_reg: pmic register that is to be used for this VR
+ */
+struct intel_pmic_info {
+	struct regulator_init_data *init_data;
+	struct regulator_dev *intel_pmic_rdev;
+	const u16 *table;
+	u16 pmic_reg;
+	u8 table_len;
+};
-- 
2.37.3

